torchdynamo/utils.py

```diff
@@ -633,3 +633,26 @@ def recompile_reasons(code):
             rpt += "No cache-limited recompilations detected.\n"
 
         return rpt
+
+
+def recursive_allclose(lhs, rhs, rtol=1e-5, atol=1e-8):
```

Comment:
is this the same as testing.same()?

Reply:
Yes, but  (A) its used in non testing code, so I figured we should have a smaller, separate impl w/o the `elif type(a).__name__ in (` stuff and (B) I ripped it off from somewhere else.

I could ostensibly consolidate the testing impl to use this one, but I'd rather not have non-testing code import testing code.

Reply:
Lets combine them.